modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/UserUtilTest.java

```diff
@@ -0,0 +1,216 @@
+import com.liferay.portal.kernel.test.util.UserGroupTestUtil;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.util.UserUtil;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Stefano Motta
+ */
+@RunWith(Arquillian.class)
+public class UserUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_user = UserTestUtil.addUser(RandomTestUtil.randomString());
+	}
+
+	@Test
+	public void testGetAllUserRoles() throws Exception {
+		List<Role> roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 1, roles.size());
+
+		_setRole();
+
+		roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 2, roles.size());
+
+		_setInheritedRole();
+
+		roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 3, roles.size());
+
+		_setOrganizationRole();
+
+		roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 4, roles.size());
+
+		_setSiteRole();
+
+		roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 5, roles.size());
+
+		_setInheritedSiteRole();
+
+		roles = UserUtil.getAllUserRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 6, roles.size());
+	}
+
+	@Test
+	public void testGetInheritedRoles() throws Exception {
+		List<Role> roles = UserUtil.getInheritedRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 0, roles.size());
+
+		_setInheritedRole();
+
+		roles = UserUtil.getInheritedRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 1, roles.size());
+	}
+
+	@Test
+	public void testGetInheritedSiteRoles() throws Exception {
+		List<Role> roles = UserUtil.getInheritedSiteRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 0, roles.size());
+
+		_setInheritedSiteRole();
+
+		roles = UserUtil.getInheritedSiteRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 1, roles.size());
+	}
+
+	@Test
+	public void testGetOrganizationsRoles() throws Exception {
+		List<Role> roles = UserUtil.getOrganizationsRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 0, roles.size());
+
+		_setOrganizationRole();
+
+		roles = UserUtil.getOrganizationsRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 1, roles.size());
+	}
+
+	@Test
+	public void testGetSiteRoles() throws Exception {
+		List<Role> roles = UserUtil.getSiteRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 0, roles.size());
+
+		_setSiteRole();
+
+		roles = UserUtil.getSiteRoles(_user);
+
+		Assert.assertEquals(roles.toString(), 1, roles.size());
+	}
```

Comment:
These tests look like are already verified with the first test, aren't they?

Reply:
The first test verify the all list but these are verifying the specific type (getSiteRoles, getOrganizationsRoles, getInheritedSiteRoles...)